File: exemplos/python/regex.py
# Para executar esse programa, no diretório onde ele se encontra rodar:
# "C:\Users\myuser\Documents\apps\python-3.9.5-embed-amd64\python" regex.py

import re

regexp = r'(\d\d)(\w)'
alvo = '11a22b33c'

print("Unico match:")
print()

resultado = re.search(regexp, alvo)
print(resultado.group())
print(resultado.group(0))
print(resultado.group(1))
print(resultado.group(2))
print(resultado.start())
print(resultado.end())

print()
print("Varios matches:")
print()

resultados = re.finditer(regexp, alvo)
for resultado in resultados:
    print("%s | %s | %s [%s,%s]" % (resultado.group(), resultado.group(1), resultado.group(2), resultado.start(), resultado.end()))

File: exemplos/ruby/regex.rb
# para executar esse programa, rodar no diretório onde ele está: ruby regex.rb

regex = /(\d\d)(\w)/
alvo = "11a22b33c"

resultado = regex.match(alvo)

print 'match inteiro: '
puts resultado[0]
print 'begin: '
puts resultado.begin 0
print 'end: '
puts resultado.end 0
puts

print 'primeiro grupo: '
puts resultado[1]
print 'begin: '
puts resultado.begin 1
print 'end: '
puts resultado.end 1
puts

print 'segundo grupo: '
puts resultado[2]
print 'begin: '
puts resultado.begin 2
print 'end: '
puts resultado.end 2
puts

print 'varios matches:'
puts
resultados = alvo.scan regex
puts resultados [0][0]
puts resultados [0][1]
puts resultados [1][0]
puts resultados [1][1]
puts resultados [2][0]
puts resultados [2][1]